kubernetes/utils/delete_from_yaml.py

```diff
@@ -0,0 +1,146 @@
+    """
+    Perform an action from a dictionary containing valid kubernetes
+    API object (i.e. List, Service, etc).
+    Input:
+    k8s_client: an ApiClient object, initialized with the client args.
+    data: a dictionary holding valid kubernetes objects
+    verbose: If True, print confirmation from the create action.
+        Default is False.
+    namespace: string. Contains the namespace to create all
+        resources inside. The namespace must preexist otherwise
+        the resource creation will fail. If the API object in
+        the yaml file already contains a namespace definition
+        this parameter has no effect.
+    Raises:
+        FailToDeleteError which holds list of `client.rest.ApiException`
+        instances for each object that failed to delete.
+    """
+    api_exceptions = []
+    try:
+        # call function delete_from_yaml_single_item 
+        delete_from_yaml_single_item(
+            k8s_client, yml_document, verbose, namespace=namespace, **kwargs
+        )
+    except client.rest.ApiException as api_exception:
+        api_exceptions.append(api_exception)
+
+    if api_exceptions:
+        raise FailToDeleteError(api_exceptions)
+
+
+def delete_from_yaml_single_item(k8s_client, yml_document, verbose=False, **kwargs):
+    # get group and version from apiVersion
+    group,_,version = yml_document["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split('.'))
+    func = "{0}{1}Api".format(group, version.capitalize())
+    k8s_api = getattr(client, func)(k8s_client)
+    kind = yml_document["kind"]
+    kind = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', kind)
+    kind = re.sub('([a-z0-9])([A-Z])', r'\1_\2', kind).lower()
+
+    if getattr(k8s_api,"delete_namespaced_{}".format(kind)):
+        # load namespace if provided in yml file
+        if "namespace" in yml_document["metadata"]:
+            namespace = yml_document["metadata"]["namespace"]
+            kwargs["namespace"] = namespace
+        # take name input of kubernetes object
+        name = yml_document["metadata"]["name"]
+        #call function to delete from namespace
+        res = getattr(k8s_api,"delete_namespaced_{}".format(kind))(
+         name=name,body=client.V1DeleteOptions(propagation_policy="Foreground", grace_period_seconds=5),**kwargs)
+
+    else:
+        # get name of object to delete
+        name = yml_document["metadata"]["name"]
+        kwargs.pop('namespace', None)
+        res = getattr(k8s_api,"delete_{}".format(kind))(
+         name=name,body=client.V1DeleteOptions(propagation_policy="Foreground", grace_period_seconds=5),**kwargs)
```

Comment:
A 200 response from a delete request doesn't guarantee the resource being deleted. We should wait until the resource is no longer visible from the server.

Reply:
How should the  e2e tests work for objects being deleted , after deletion if they are search (the object) it gets timed out resulting in an error  ,however  in the create_from_yaml tests  in the end all resources are being deleted using multiple separate commands   could we delete those using the delete_from_yaml method and that would be sufficient?

Reply:
/assign @yliaog

Reply:
in e2e, after delete, try listing the object again, if it does not exist anymore, it means the object is successfully deleted.